src/Miscellaneous/PostJumper.coffee

```diff
@@ -0,0 +1,52 @@
+PostJumper = 
+  init: ->
+    @capcodeMap  = new Map()
+    @uniqueIDMap = new Map()
+    return unless g.VIEW in ['index', 'thread']
+
+    Callbacks.Post.push
+      name: 'Post Jumper'
+      cb:   @node
+
+  node: ->
+    if @nodes.uniqueIDRoot and Conf['Unique ID Navigation'] and not @nodes.uniqueIDJumperRoot
+      PostJumper.addButtons @,'uniqueID'
+
+    if @nodes.capcode and Conf['Capcode Navigation'] and not @nodes.capcodeJumperRoot
+      PostJumper.addButtons @,'capcode'
+
+  addButtons: (post,type) ->
+    value = post.nodes[type].innerText
+    buttons = PostJumper.makeButtons type+'Jumper'
+    $.after post.nodes[type+(if type is 'capcode' then '' else 'Root')], buttons
+    $.on buttons.firstChild, 'click', PostJumper.buttonClick post,type,-1
+    $.on buttons.lastChild, 'click', PostJumper.buttonClick post,type,1
+    if not PostJumper[type+'Map'].has value
+      PostJumper[type+'Map'].set value, []
+    PostJumper[type+'Map'].get(value).push post.nodes.quote.innerText  
```

Comment:
This will be out of order if the posts are created out of order. An example would be expanding a thread in the index; the OP and the visible replies are created before the intermediate replies.

>`post.nodes.quote.innerText`
This should use `post.ID`.

Also `PostJumper.maps[type]` might be nicer than `PostJumper[type+'Map']`.